### PR TITLE
Slight area code cleanup.

### PR DESCRIPTION
**Pull request by DominikPanic**
Doesn't change the functionality of anything nor makes any future port harder, but cleans up what we already have.

### CODE REVIEW

File: code/game/area/almayer.dm [hunk 1]
```diff
@@ -374,37 +374,4 @@
...
-/area/almayer/evacuation/stranded/pod3
-/area/almayer/evacuation/stranded/pod4
-/area/almayer/evacuation/stranded/pod5
-/area/almayer/evacuation/stranded/pod6
-/area/almayer/evacuation/stranded/pod7
-/area/almayer/evacuation/stranded/pod8
-/area/almayer/evacuation/stranded/pod9
-/area/almayer/evacuation/stranded/pod10
-/area/almayer/evacuation/stranded/pod11
+	requires_power = 0
```
Comment (spookydonut) [inline code/game/area/almayer.dm]:
```suggestion
	requires_power = FALSE
```

File: code/game/area/area.dm [hunk 2]
```diff
@@ -38,14 +84,59 @@
...
+	SEND_SIGNAL(M, COMSIG_EXIT_AREA, src) //The atom that exits the area
+
+
 /area/proc/reg_in_areas_in_z()
-	if(contents.len)
+	if(length(contents))
 		var/list/areas_in_z = SSmapping.areas_in_z
 		var/z
-		for(var/i in 1 to contents.len)
+		for(var/i in 1 to length(contents))
```
Comment (spookydonut) [inline code/game/area/area.dm]:
```suggestion
		for(var/i in contents)
```

File: code/game/area/area.dm [hunk 3]
```diff
@@ -38,14 +84,59 @@
...
 			var/atom/thing = contents[i]
```
Comment (spookydonut) [inline code/game/area/area.dm]:
```suggestion
			var/atom/thing = i
```

File: code/game/area/area.dm [hunk 4]
```diff
@@ -146,27 +233,37 @@
 		return TRUE
 	return FALSE
 
+
 /area/proc/air_doors_close()
```
Comment (spookydonut) [inline code/game/area/area.dm]: don't think anything uses these

Comment (DominikPanic) [reply code/game/area/area.dm]: The fire alarms do

File: code/game/area/area.dm [hunk 5]
```diff
@@ -38,15 +84,60 @@
...
+	SEND_SIGNAL(src, COMSIG_AREA_ENTERED, AM)
+	SEND_SIGNAL(AM, COMSIG_ENTER_AREA, src) //The atom that enters the area
+	if(!isliving(AM))
+		return
+
+	var/mob/living/L = AM
+	if(!L.ckey)
+		return
+
+	if(!(L.client && (L.client.prefs.toggles_sound & SOUND_AMBIENCE)))
```
Comment (Rohesie) [inline code/game/area/area.dm]:
```suggestion
	if(!L.client || !(L.client.prefs.toggles_sound & SOUND_AMBIENCE))
```

File: code/game/area/area.dm [hunk 6]
```diff
@@ -38,14 +84,59 @@
...
+
+
+/area/Exited(atom/movable/M)
+	SEND_SIGNAL(src, COMSIG_AREA_EXITED, M)
+	SEND_SIGNAL(M, COMSIG_EXIT_AREA, src) //The atom that exits the area
+
+
 /area/proc/reg_in_areas_in_z()
-	if(contents.len)
+	if(length(contents))
```
Comment (spookydonut) [inline code/game/area/area.dm]:
```suggestion
	if(!length(contents)) return
```
erli retern

**Pull request merged**